src/main/haskell/kore/src/Kore/AST/Lens.hs

```diff
@@ -0,0 +1,304 @@
+{-|
+Module      : Kore.AST.Lens
+Description : Lenses into Kore patterns
+Copyright   : (c) Runtime Verification, 2018
+License     : NCSA
+Maintainer  : phillip.harris@runtimeverification.com
```

Comment:
These were originally part of `Kore.ASTUtils.SmartConstructors` (which Phillip wrote). I moved them into their own module because I was removing the rest of that module and I thought it would be better to delete it than leave a module called "SmartConstructors" which contained no smart constructors.